src/Microsoft.Dnx.Runtime.Internals/RuntimeEnvironmentExtensions.cs

```diff
@@ -27,5 +30,45 @@ public static string GetShortVersion(this IRuntimeEnvironment env)
         {
             return $"{env.RuntimeType}-{env.RuntimeArchitecture}-{env.RuntimeVersion}";
         }
+
+        public static string GetRuntimeIdentifier(this IRuntimeEnvironment env)
+        {
+            if (string.Equals(env.RuntimeType, RuntimeTypes.Mono, StringComparison.Ordinal))
+            {
+                return "mono";
+            }
+
+            string os = env.OperatingSystem;
+            string ver = env.OperatingSystemVersion;
+            if (string.Equals(os, RuntimeOperatingSystems.Windows, StringComparison.Ordinal))
+            {
+                os = "win";
+
+                // Convert 6.x to the correct branding version for the RID
+                var parsedVersion = Version.Parse(ver);
+                if(parsedVersion.Major == 6 && parsedVersion.Minor == 1)
+                {
+                    ver = "7";
+                }
+                else if(parsedVersion.Major == 6 && parsedVersion.Minor == 2)
+                {
+                    ver = "8";
+                }
+                else if(parsedVersion.Major == 6 && parsedVersion.Minor == 3)
+                {
+                    ver = "81";
+                }
+                else if(parsedVersion.Major == 10 && parsedVersion.Minor == 0)
+                {
+                    ver = "10";
+                }
+            }
+            else
+            {
+                os = os.ToLower(); // Just use the lower-case full name of the OS as the RID OS
```

Comment:
This will end up being wrong on certain distros right? I guess we're assuming that when @moozzyk makes the changes to flow the relevant information,  this will fix itself right?